code/player/Player.cs

```diff
@@ -31,39 +42,25 @@ public override void Respawn()
             Controller = new WalkController();
             Animator = new StandardPlayerAnimator();
             Camera = new FirstPersonCamera();
+            Inventory = new Inventory(this);
 
             EnableAllCollisions = true;
             EnableDrawing = true;
             EnableHideInFirstPerson = true;
             EnableShadowInFirstPerson = true;
 
-            base.Respawn();
-        }
-
-        public bool IsSpectator
-        {
-            get => (Camera is SpectateCamera);
-        }
+            Role = RoleType.None;
+            Credits = 0;
 
-        public void MakeSpectator(Vector3 position = default)
-        {
-            EnableAllCollisions = false;
-            EnableDrawing = false;
-            Controller = null;
-            Camera = new SpectateCamera
-            {
-                DeathPosition = position,
-                TimeSinceDied = 0
-            };
+            base.Respawn();
         }
 
         public override void OnKilled()
         {
             base.OnKilled();
 
-            CreateBodyOnServer(_lastDamageInfo.Force, GetHitboxBone(_lastDamageInfo.HitboxIndex));
+            BecomePlayerCorpseOnServer( _lastDamageInfo.Force, GetHitboxBone( _lastDamageInfo.HitboxIndex ) );
```

Comment:
why did you add spaces to the brackets?

Reply:
No reason, `Hidden` had a completely functioning body system so seemed reasonable to make use of that as a base. Just sloppy copy and paste.

Reply:
Oh, I was confused. While reading it I thought Alf made a code suggestion. My bad, I'm tired :D

Reply:
@Alf21 I'll see if this is possible, but, I'm assuming FP had some reason why they didn't just reuse the player model somehow.

Reply:
Thought the player is not directly connected to the Client as you're using Client.Pawn, so that you can move every model / player assigned to Client.Pawn ^^

Reply:
Oh, if I understand correctly, you mean assigning the `body` to `Client.Pawn`? If so, I don't think that would work since we still technically have a `TTTPlayer` as `Client.Pawn` even after they die (as a spectator).